### ISSUE TITLE
0-db can't compile/run on MacOS

### ISSUE THREAD

**Issue by GlenDC** [threefoldtech/0-db#1]
As documented, and tested, 0-db cannot be compiled on MacOS. If you do try to compile (using `make`, you'll get following error:

```
cc -g -W -Wall -O2 -msse4.2 -c redis.c
redis.c:12:10: fatal error: 'sys/epoll.h' file not found
#include <sys/epoll.h>
         ^
1 error generated.
make[1]: *** [redis.o] Error 1
```

The issue is that `epoll` isn't supported by MacOS. Instead we should go for `kqueue` (as a replacement for `epoll`, which will also have to be used for FreeBSD. For Linux we can keep using `epoll`.

**Comment by GlenDC** [comment]
This bug has to be fixed in order to start development of 0-nbd (as tracked in https://github.com/zero-os/0-nbd/issues/3), as my dev machine is MacOS.

**Comment by GlenDC** [comment]
Fixed in fc5c1284792d894526147e16277d31906c6de042 and documented in 29f343ddfc8230b981f8735598a72d59d8a822aa.
